### PR TITLE
My Jetpack: make card primary button links when applicable 

### LINKED ISSUE #31616: [Flagship plugin] My Jetpack: buttons in cards should be links, as the action is directing to another page

### PR DESCRIPTION
**Pull request by monsieur-z**
Fixes #31616 #31619

## Proposed changes:

In the _My Jetpack_ section, activating the primary button of product cards often directs users to a different page. They're effectively links rather than buttons but are always implemented with the `button` tag. Using the proper semantics is important: it allows, for instance, screen reader users to better identify links present on the page.

This PR implements links when applicable.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pedMtX-Rg-p2

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Spin up a test site with this branch
* Visit the _My Jetpack_ page
* For each possible action (_Activate_, _Manage_, _Purchase_, _Start for Free_, _Install_), check that the behaviour is unchanged

Screenshot 2023-06-27 at 2 11 11 PM

### CODE REVIEW

File: projects/packages/my-jetpack/_inc/components/connected-product-card/index.jsx
Comment (monsieur-z) [inline projects/packages/my-jetpack/_inc/components/connected-product-card/index.jsx]: `manageUrl` is retrieved in `ActionButton` instead.

File: projects/packages/my-jetpack/_inc/components/connected-product-card/index.jsx
Comment (monsieur-z) [inline projects/packages/my-jetpack/_inc/components/connected-product-card/index.jsx:68]: Not necessary when using links.

File: projects/packages/my-jetpack/_inc/components/connected-product-card/index.jsx
Comment (monsieur-z) [inline projects/packages/my-jetpack/_inc/components/connected-product-card/index.jsx:38]: Set `href`s in `ActionButton` directly instead.

File: projects/packages/my-jetpack/_inc/components/product-card/index.jsx
Comment (monsieur-z) [inline projects/packages/my-jetpack/_inc/components/product-card/index.jsx:258]: Forcing the card to be a `div` since having an interactive element (`a` or `button`) inside another is neither valid HTML nor accessible.

File: projects/packages/my-jetpack/_inc/components/product-card/action-buton.jsx
Comment (andrii-lysenko) [inline projects/packages/my-jetpack/_inc/components/product-card/action-buton.jsx]: Not a blocker for me, but maybe we could pass this URL as a parameter, to get more flexibility. Not a requirenment though, just a feeling.

File: projects/packages/my-jetpack/_inc/components/product-card/index.test.jsx
Comment (andrii-lysenko) [inline projects/packages/my-jetpack/_inc/components/product-card/index.test.jsx:1]: We don't add any tests instead?

**Review by andrii-lysenko — DISMISSED** [review]
It looks good and tests well 👍 . There are some small questions, but they are not blocking for me.

File: projects/packages/my-jetpack/_inc/components/product-card/index.test.jsx
Comment (monsieur-z) [reply projects/packages/my-jetpack/_inc/components/product-card/index.test.jsx:1]: I don't think it's necessary to add tests, at least to replace these ones. `onManage` and `onAdd` have been removed in favor of using `href`, which ultimately is passed to an `a` tag.

File: projects/packages/my-jetpack/_inc/components/product-card/action-buton.jsx
Comment (monsieur-z) [reply projects/packages/my-jetpack/_inc/components/product-card/action-buton.jsx]: I don't see what you have in mind. Could you add an example?

**Comment by monsieur-z** [comment]
Fixed conflicts with https://github.com/Automattic/jetpack/pull/31629/

File: projects/packages/my-jetpack/_inc/components/product-card/action-buton.jsx
Comment (andrii-lysenko) [reply projects/packages/my-jetpack/_inc/components/product-card/action-buton.jsx]: It's actually not so crucial, you can disregard this comment

**Review by andrii-lysenko — DISMISSED** [review]
Still LGTM after merge. Works great 👍

**Review by jeherve — DISMISSED** [review]
This tests well for me. 🚢

**Review by andrii-lysenko — APPROVED** [review]

**Pull request merged**
